tools/git-hooks/pre-commit

```diff
@@ -0,0 +1,45 @@
+#! /bin/bash
+
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
+# SPDX-FileContributor: Mathias Kraus
+
+# This script checks for trailing whitespaces on the modified file and runs 'cargo fmt --check'
+
+set -eu
```

Comment:
What does this little thing do?

Reply:
The location. In my case European Union. It's `us` if you want to use the script from the United States ;)

- e  Exit immediately if a command exits with a non-zero status
- u  Treat unset variables as an error when substituting